genesis/model/resolver/global.go

```diff
@@ -57,6 +57,7 @@ func (r *globalResolver) GetObject(reference interface{}, cls interface{}) (inte
 	if proxy.GetClassID() != classID {
 		return nil, fmt.Errorf("instance class is not `%s`", classID)
 	}
+	proxy.SetReference(reference.(object.Reference))
```

Comment:
Here `ref, ok := reference.(object.Reference)` you already got reference as object.Reference, so you can do just `proxy.SetReference(ref)` now